### PR TITLE
Better link parsing

### PR DESCRIPTION
**Pull request by Emporophobe**
Fixed bad link parsing for articles with slashes in the title using URL encoding.

### CODE REVIEW

File: tests/Tester.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def run_tests():
     log.info("Results will be stored in {}".format(results_file_path))
 
     with open(results_file_path, 'w', newline='') as f:
-        writer = csv.writer(f, delimiter='\t')
+        writer = csv.writer(f, delimiter='\t' )
```
Comment (libreite) [inline tests/Tester.py]: Why?

Comment (kj800x) [inline tests/Tester.py]: Weird space?

File: src/apis/WikipediaApi.py [hunk 2]
```diff
@@ -57,10 +58,10 @@ def get_random_text_and_links(self):
     def extract_text_and_links(page):
         response = BeautifulSoup(page.text, 'html.parser')
         links = response.find_all('a')
-        # TODO: This link parsing needs to be fixed to get all the links we want and nothing else
-        unique_links = set([link["href"].split("/")[-1].split("#")[0]
+        prefix = "./"
```
Comment (kj800x) [inline src/apis/WikipediaApi.py]: Shouldn't the prefix be './wiki/'

Unless I'm not understanding the code?

Comment (Emporophobe) [reply src/apis/WikipediaApi.py]: The html returned from the API is different from just viewing the html on the pages themselves; internal links are all of the format `./banana`.

File: tests/Tester.py (hunk 1 above)
Comment (Emporophobe) [reply tests/Tester.py]: 🤷‍♂️

**Pull request merged**